### DISCUSSION TITLE
Constructor functions - how to implement?

### DISCUSSION THREAD

**Discussion by TangkasOka** [louthy/language-ext#1025] [Q&A] [answered]
Hi,

This is probably an RTFM question, but I'm not sure where to look for the FM. :-)

I'm interested in implementing constructor functions for creating objects without using "new".  That is, using the same semantics in new classes that I write as is done in LanguageExt.

Like what is done in LanguageExt for "Seq()".

Are there any tips on how this is done?  Either documentation or code.

Thanks, Charlie

**Comment by louthy** [comment, marked as answer]
Just create static methods in static classes, then do:

    using static YourClass

That will make them act like top level functions

**Reply by TangkasOka** [reply]
Thanks!
